[CI] RandomizedTimeSeriesIT testRateGroupByNothing failing

**Build Scans:**
- [elasticsearch-intake #33413 / part3](https://gradle-enterprise.elastic.co/s/mtayq7yb7qsmo)
- [elasticsearch-pull-request #114820 / part-3-fips](https://gradle-enterprise.elastic.co/s/bh445hzkujp3w)
- [elasticsearch-periodic-platform-support #11711 / debian-13_platform-support-unix](https://gradle-enterprise.elastic.co/s/2qp6pdr5guz4w)
- [elasticsearch-pull-request #114047 / part-3](https://gradle-enterprise.elastic.co/s/jkskz46ipoa2u)
- [elasticsearch-periodic-platform-support #11701 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/xgqlq4tbjuh6e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT.testRateGroupByNothing" -Dtests.seed=627E277B80832507 -Dtests.locale=teo-KE -Dtests.timezone=Europe/Malta -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRateGroupByNothing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed. Agg: RATE | Failed windows: 1 windows(1.6666666666666667%):
Failed for row:
[17, 34.86020237922169, 9.190628178521987, 1.6898218708870891, 2025-07-31T09:06:00.000Z]
Wanted: RateStats[count=16, max=RateRange[lower=6.947659347861283, upper=218.10574436779143], avg=RateRange[lower=4.8546344760157885, upper=33.83577416786363], min=RateRange[lower=0.49849344299563914, upper=4.185503299711496], sum=RateRange[lower=77.67415161625262, upper=541.3723866858181]]
Exception: 
Expected: <16L>
     but: was <17L>
```

**Issue Reasons:**
- [main] 5 failures in test testRateGroupByNothing (0.7% fail rate in 746 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.4% fail rate in 491 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,9 +354,6 @@ tests:
 - class: org.elasticsearch.repositories.blobstore.testkit.analyze.RepositoryAnalysisFailureIT
   method: testFailsOnCopyAfterWrite
   issue: https://github.com/elastic/elasticsearch/issues/140819
-- class: org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT
-  method: testRateGroupByNothing
-  issue: https://github.com/elastic/elasticsearch/issues/139639
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -50,6 +50,7 @@ import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcke
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.closeTo;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.lessThan;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
@@ -418,9 +419,9 @@ public class RandomizedTimeSeriesIT extends AbstractEsqlIntegTestCase {
     ) {
         String timeseriesId = timeseries.getFirst().v1();
         var referenceTuple = isLowerBoundary ? timeseries.getFirst().v2() : timeseries.getLast().v2();
-        if (referenceTuple.v1().toEpochMilli() % (secondsInWindow * 1000L) == 0) {
+        if (isLowerBoundary && referenceTuple.v1().toEpochMilli() % (secondsInWindow * 1000L) == 0) {
             // The reference tuple is already on the boundary.
-            return false;
+            return true;
         }
         Tuple<Instant, Double> otherTuple = null;
         long otherTimestamp = 0;

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -50,7 +50,6 @@ import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcke
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.closeTo;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.lessThan;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;

